src/weights.jl

```diff
@@ -193,9 +195,87 @@ pweights(vs::RealArray) = ProbabilityWeights(vec(vs))
     end
 end
 
+# NOTE: A custom constructor is needed because exponential weights must
+# sum to 1, so we want to enforce that on construction of the type
+@weights ExponentialWeights false
+
+"""
+    ExponentialWeights(vs)
+
+Construct an `ExponentialWeights` vector with weight values `vs`, which must sum to 1.
```

Comment:
Is the sum to 1 necessary?

Reply:
Not strictly necessary, but the `varcorrection` optimization is kind of a nice property.

Reply:
I'll admit I'm not entirely sure how `varcorrection` gets used, but normalizing the weights is unexpected (as far as I understand how people use exponential weights).